src/Core/maps/src/Maps/Handlers/Map/MapHandler.iOS.cs

```diff
@@ -0,0 +1,170 @@
+				_mapClickedGestureRecognizer.Dispose();
+				_mapClickedGestureRecognizer = null;
+			}
+
+			if (platformView is MauiMKMapView customMKMapView)
+				customMKMapView.LayoutSubviewsFired -= CustomMKMapViewLayoutSubviewsFired;
+
+
+			platformView.RegionChanged -= MkMapViewOnRegionChanged;
+			//platformView.OverlayRenderer = null;
+
+			var mapsPinsItemsSource = (ObservableCollection<IMapPin>)VirtualView.Pins;
+			mapsPinsItemsSource.CollectionChanged -= OnPinCollectionChanged;
+
+			// This handler is done with the MKMapView; we can put it in the pool
+			// for other rendererers to use in the future
+			MapPool.Add(platformView);
+		}
+
+		public static void MapMapType(IMapHandler handler, IMap map)
+		{
+			switch (map.MapType)
+			{
+				case MapType.Street:
+					handler.PlatformView.MapType = MKMapType.Standard;
+					break;
+				case MapType.Satellite:
+					handler.PlatformView.MapType = MKMapType.Satellite;
+					break;
+				case MapType.Hybrid:
+					handler.PlatformView.MapType = MKMapType.Hybrid;
+					break;
+			}
+		}
+
+		public static void MapIsShowingUser(IMapHandler handler, IMap map)
+		{
+
+#if !MACCATALYST
+			if (map.IsShowingUser)
+			{
+				MapHandler? mapHandler = handler as MapHandler;
+				mapHandler?._locationManager?.RequestWhenInUseAuthorization();
+			}
+#endif
+			handler.PlatformView.ShowsUserLocation = map.IsShowingUser;
+		}
+
+		public static void MapHasScrollEnabled(IMapHandler handler, IMap map)
+		{
+			handler.PlatformView.ScrollEnabled = map.HasScrollEnabled;
+		}
+
+		public static void MapHasTrafficEnabled(IMapHandler handler, IMap map)
+		{
+			handler.PlatformView.ShowsTraffic = map.HasTrafficEnabled;
+		}
+
+		public static void MapHasZoomEnabled(IMapHandler handler, IMap map)
+		{
+			handler.PlatformView.ZoomEnabled = map.HasZoomEnabled;
+		}
+
+		public static void MapMoveToRegion(IMapHandler handler, IMap map, object? arg)
+		{
+			MapSpan? newRegion = arg as MapSpan;
+			if (newRegion != null)
+				(handler as MapHandler)?.MoveToRegion(newRegion, true);
+		}
+
+		//protected virtual MKOverlayRenderer GetViewForOverlay(MKMapView mapview, IMKOverlay overlay)
+		//{
+		//	//switch (overlay)
+		//	//{
+		//	//	case MKPolyline polyline:
+		//	//		return GetViewForPolyline(polyline);
+		//	//	case MKPolygon polygon:
+		//	//		return GetViewForPolygon(polygon);
+		//	//	case MKCircle circle:
+		//	//		return GetViewForCircle(circle);
+		//	//}
+
+		//	return null;
+		//}
+
+		void CustomMKMapViewLayoutSubviewsFired(object? sender, EventArgs e)
+		{
+			//MoveToRegion(VirtualView.LastMoveToRegion, false);
+		}
+
+		void OnMapClicked(UITapGestureRecognizer recognizer)
+		{
+			var tapPoint = recognizer.LocationInView(PlatformView);
+			var tapGPS = PlatformView.ConvertPoint(tapPoint, PlatformView);
+			VirtualView.Clicked(new Devices.Sensors.Location(tapGPS.Latitude, tapGPS.Longitude));
+		}
+
+		void MkMapViewOnRegionChanged(object? sender, MKMapViewChangeEventArgs e)
```

Comment:
- Can this be static?
- A delegate might be better?
- Or you could subscribe/unsubscribe from this inside "MovedToWindow" on the MauiMapView.  

iOS has trouble collecting things with handlers like this.